File: cpp/0001-two-sum.cpp
#include <unordered_map>
#include <vector>

using uint = unsigned;

class Solution {
    public:
        [[nodiscard]] auto twoSum(
            const std::vector<int> &nums,
            const int target
        ) const noexcept -> std::vector<int> {
            auto d = std::unordered_map<int, int>{};

            for (auto i = 0U; i < nums.size(); ++i) {
                const auto n = nums[i];
                const auto diff = target - n;

                if (d.contains(diff)) {
                    return std::vector{d[diff], static_cast<int>(i)};
                }

                d[n] = static_cast<int>(i);
            }

            return std::vector<int>{};
        }
};

File: py/0001-two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d = dict[int, int]()

        for i, n in enumerate(nums):
            diff = target - n
            if diff in d:
                return [d[diff], i]
            d[n] = i

        return []
